eng/common/scripts/Helpers/Metadata-Helpers.ps1

```diff
@@ -47,7 +47,30 @@ function GetMsAliasFromGithub ($TenantId, $ClientId, $ClientSecret, $GithubUser)
     return $null
 }
 
-function GetPrimaryCodeOwner ($TargetDirectory) 
+function GetAllGithubUsers ([string]$TenantId, [string]$ClientId, [string]$ClientSecret)
+{
+    # API documentation (out of date): https://github.com/microsoft/opensource-management-portal/blob/main/docs/api.md
+    $OpensourceAPIBaseURI = "https://repos.opensource.microsoft.com/api/people/links"
+
+    $Headers = @{
+        "Content-Type" = "application/json"
+        "api-version" = "2019-10-01"
+    }
+
+    try {
+        $opsAuthToken = Generate-AadToken -TenantId $TenantId -ClientId $ClientId -ClientSecret $ClientSecret
+        $Headers["Authorization"] = "Bearer $opsAuthToken"
+        Write-Host "Fetching all github alias links"
+        $resp = Invoke-RestMethod $OpensourceAPIBaseURI -Method 'GET' -Headers $Headers -MaximumRetryCount 3
```

Comment:
Does this need any paging?

Reply:
From what I can see, no. These [docs](https://github.com/microsoft/opensource-management-portal/blob/main/docs/api.md#get-all-linked-users) are slightly out of date but don't mention it. Also, given that the API call takes 15 seconds and returns a 15mb object, I'm betting that if there was paging it wouldn't behave like that. Hence the caching functionality added.